servers/zms/src/main/java/com/yahoo/athenz/zms/utils/ZMSUtils.java

```diff
@@ -271,20 +271,27 @@ public static Principal.Type principalType(final String memberName, final String
     public static boolean isUserDomainPrincipal(final String memberName, final String userDomainPrefix,
             final List<String> addlUserCheckDomainPrefixList) {
 
-        if (memberName.startsWith(userDomainPrefix)) {
+        if (memberName.startsWith(userDomainPrefix) && StringUtils.countMatches(memberName, '.') == 1) {
             return true;
         }
 
         if (addlUserCheckDomainPrefixList != null) {
             for (String prefix : addlUserCheckDomainPrefixList) {
-                if (memberName.startsWith(prefix)) {
+                if (memberName.startsWith(prefix) && StringUtils.countMatches(memberName, '.') == 1) {
                     return true;
                 }
             }
         }
 
         return false;
     }
+    
+    public static boolean isHeadlessUserDomainPrincipal(final String memberName, final String headlessUserDomainPrefix) {
+        if (memberName.startsWith(headlessUserDomainPrefix) && StringUtils.countMatches(memberName, '.') == 1) {
```

Comment:
we should simplify this with just a single return statement:

```
return memberName.startsWith(headlessUserDomainPrefix) && StringUtils.countMatches(memberName, '.') == 1;
```

Reply:
fixed!